stdlib/3/html/parser.pyi

```diff
@@ -15,7 +15,7 @@ class HTMLParser(ParserBase):
     def get_starttag_text(self) -> str: ...
 
     def handle_starttag(self, tag: str,
-                        attrs: List[Tuple[str, str]]) -> None: ...
+                        attrs: List[Tuple[str, Optional[str]]]) -> None: ...
```

Comment:
This can be problematic because List is invariant, so code that passes a `List[Tuple[str, str]]` will no longer typecheck. Can the List be a Sequence or Iterable instead?

Reply:
While this is true, IIRC `handle_starttag()` (and the other callbacks) are usually not called by user code, but by XML parsers. User code overrides these methods to handle parser events. In a way, this is the opposite of how APIs normally work. Therefore I think using a concrete type is more convenient in this case. But there certainly is a case to be made for non-mutable types as well.